api/sso_test.go

```diff
@@ -515,6 +516,128 @@ func (ts *SSOTestSuite) TestAdminDeleteSSOProvider() {
 	}
 }
 
+func (ts *SSOTestSuite) TestSingleSignOn() {
+	providers := []struct {
+		ID      string
+		Request map[string]interface{}
+	}{
+		{
+			Request: map[string]interface{}{
+				"type":         "saml",
+				"metadata_xml": validSAMLIDPMetadata("https://accounts.google.com/o/saml2?idpid=EXAMPLE-A"),
+			},
+		},
+		{
+			Request: map[string]interface{}{
+				"type": "saml",
+				"domains": []string{
+					"example.com",
+				},
+				"metadata_xml": validSAMLIDPMetadata("https://accounts.google.com/o/saml2?idpid=EXAMPLE-B"),
+			},
+		},
+	}
+
+	for i, example := range providers {
+		body, err := json.Marshal(example.Request)
+		require.NoError(ts.T(), err)
+
+		req := httptest.NewRequest(http.MethodPost, "http://localhost/admin/sso/providers", bytes.NewBuffer(body))
+		req.Header.Set("Authorization", "Bearer "+ts.AdminJWT)
+		w := httptest.NewRecorder()
+
+		ts.API.handler.ServeHTTP(w, req)
+
+		response, err := io.ReadAll(w.Body)
+		require.NoError(ts.T(), err)
+
+		var payload struct {
+			ID string `json:"id"`
+		}
+
+		require.NoError(ts.T(), json.Unmarshal(response, &payload))
+
+		providers[i].ID = payload.ID
+	}
+
+	examples := []struct {
+		Code    int
+		Request map[string]interface{}
+		URL     string
+	}{
+		{
+			Request: map[string]interface{}{
```

Comment:
/nit would be nice to have test case descriptions

Reply:
+1 on this - would make the test cases easier to understand!

Reply:
+1 on this - would make the test cases easier to understand!